.appveyor.yml

```diff
@@ -1,4 +1,4 @@
-version: 4.1.0.99.{build}
+version: 4.2.0.{build}
```

Comment:
Just want to make sure why we are using 4.2.0 now instead of 4.2.0.99 to differentiate between released version and the one built from source?

Reply:
Like we've done for previous releases, after this that'll get changed to `4.2.0.99` in a follow-up PR.

For example: #6090

This version doesn't affect any artifacts that are delivered to users or anything. Just the way builds are organized in the AppVeyor UI.

https://ci.appveyor.com/project/guolinke/lightgbm/history

<img width="1045" alt="image" src="https://github.com/microsoft/LightGBM/assets/7608904/1e353b79-7e47-4398-bddd-c08cd22bce54">

Doing this on the release PR ensures these builds are identifiable in the future as belong to the `4.2.0` release.

> differentiate between released version and the one built from source

The commit produced when we merge this PR will be the released version.